### PR TITLE
A working subset of #1788

### PR DESCRIPTION
**Pull request by g1itch**
Hello!

This is a subset of #1788, the tests that are ready (+4 tests, including those for base58 and varint) and corresponding changes in the `protocol` module.

### CODE REVIEW

**Comment by g1itch** [comment]
There is also a test for base58 in #1796: 7f213f6

File: src/tests/test_addresses.py [hunk 1]
```diff
@@ -33,3 +33,21 @@ def test_encode(self):
         self.assertEqual(
             sample_address,
             addresses.encodeAddress(2, 1, unhexlify(sample_ripe)))
+
+    def test_base58(self):
+        """Check Base58 encoding and decoding"""
+        self.assertEqual(
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:39]: - these strings and numbers should be in `samples.py`
- to improve coverage, encode needs to test `0`, and decode needs to test an invalid character

File: src/tests/test_packets.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self.assertEqual(data, unhexlify('feffffffff'))
+        data = addresses.encodeVarint(4294967296)
+        self.assertEqual(data, unhexlify('ff0000000100000000'))
+        data = addresses.encodeVarint(18446744073709551615)
+        self.assertEqual(data, unhexlify('ffffffffffffffffff'))
+
+        with self.assertRaises(addresses.varintEncodeError):
+            addresses.encodeVarint(18446744073709551616)
+
+        value, lenght = addresses.decodeVarint(b'\xfeaddr')
```
Comment (PeterSurda) [inline src/tests/test_packets.py:38]: `length`, not `lenght`

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Mostly ok, just minor cleanup pls.

File: src/tests/test_addresses.py [hunk 3]
```diff
@@ -17,19 +23,40 @@ def test_decode(self):
...
+        self.assertEqual(addresses.decodeBase58('1'), 0)
+        self.assertEqual(addresses.decodeBase58('!'), 0)
+        self.assertEqual(
+            addresses.decodeBase58(sample_addr4), sample_daddr4_512)
+        self.assertEqual(
+            addresses.decodeBase58(sample_addr3), sample_daddr3_512)
+
+        self.assertEqual(addresses.encodeBase58(1), '2')
+        # infinite loop!
+        # self.assertEqual(addresses.encodeBase58(-1), '')
```
Comment (g1itch) [inline src/tests/test_addresses.py:58]: @PeterSurda what do you think `addresses.encodeBase58()` should do in such case? Raise an exception, return None, or drop the sign?

File: src/tests/test_addresses.py [hunk 4]
```diff
@@ -17,19 +23,40 @@ def test_decode(self):
...
             addresses.encodeAddress(2, 1, unhexlify(sample_ripe)))
+        ripe = unhexlify(sample_deterministic_ripe)
+        self.assertEqual(
+            addresses.encodeAddress(3, 1, ripe),
+            'BM-%s' % addresses.encodeBase58(sample_daddr3_512))
+
+    def test_base58(self):
+        """Check Base58 encoding and decoding"""
+        self.assertEqual(addresses.decodeBase58('1'), 0)
+        self.assertEqual(addresses.decodeBase58('!'), 0)
```
Comment (g1itch) [inline src/tests/test_addresses.py:50]: This is the invalid character.

File: src/tests/test_addresses.py (hunk 3 above)
Comment (PeterSurda) [reply src/tests/test_addresses.py:58]: Good catch, I'd go with `return None`.

File: src/tests/test_addresses.py [hunk 5]
```diff
@@ -17,19 +23,40 @@ def test_decode(self):
...
+        self.assertEqual(
+            addresses.decodeBase58(sample_addr4), sample_daddr4_512)
+        self.assertEqual(
+            addresses.decodeBase58(sample_addr3), sample_daddr3_512)
+
+        self.assertEqual(addresses.encodeBase58(1), '2')
```
Comment (PeterSurda) [inline src/tests/test_addresses.py:56]: still missing test for `0`:
`self.assertEqual(addresses.encodeBase58(0), '1')`

**Review by PeterSurda — CHANGES_REQUESTED** [review]
Still `encodeBase58(0)` is missing. Fixing negative can be fixed either in this PR or a separate one.

File: src/tests/test_addresses.py (hunk 5 above)
Comment (g1itch) [reply src/tests/test_addresses.py]: It failed when I wrote it first time, maybe I misspelled something

**Comment by g1itch** [comment]
> Fixing negative can be fixed either in this PR or a separate one.

It's a simple mechanical fix. I prefer to merge complete tests, not stubs.

**Review by PeterSurda — APPROVED** [review]

**Pull request merged**